src/App.tsx

```diff
@@ -1,15 +1,53 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [chosenPerson, setChosenPerson] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+
+  const onQueryChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (!query && !e.target.value.trim()) {
+      return;
+    }
```

Comment:
According to the checklist, the filter function should not be called if the user only entered spaces. In your current implementation, the filter will still be called if the previous query was not empty. To fix this, you should check if the trimmed value of the new query is empty and if so, don't call the filter function.